examples/bulk_imports/main.go

```diff
@@ -0,0 +1,119 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/joho/godotenv"
+
+	"github.com/hatchet-dev/hatchet/pkg/client"
+	"github.com/hatchet-dev/hatchet/pkg/worker"
+)
+
+type userCreateEvent struct {
+	Username string            `json:"username"`
+	UserID   string            `json:"user_id"`
+	Data     map[string]string `json:"data"`
+}
+
+type stepOneOutput struct {
+	Message string `json:"message"`
+}
+
+func main() {
+	err := godotenv.Load()
+	if err != nil {
+		panic(err)
+	}
+
+	// interrupt := cmdutils.InterruptChan()
+
+	_, err = run()
+	if err != nil {
+		panic(err)
+	}
+
+	// <-interrupt
+
+}
+
+func run() (func() error, error) {
+	c, err := client.New()
+
+	if err != nil {
+		return nil, fmt.Errorf("error creating client: %w", err)
+	}
+
+	w, err := worker.NewWorker(
+		worker.WithClient(
+			c,
+		),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error creating worker: %w", err)
+	}
+
+	testSvc := w.NewService("test")
+
+	err = testSvc.RegisterWorkflow(
+		&worker.WorkflowJob{
+			On:          worker.Events("user:create:bulk"),
+			Name:        "bulk",
+			Description: "This runs after an update to the user model.",
+			Steps: []*worker.WorkflowStep{
+				worker.Fn(func(ctx worker.HatchetContext) (result *stepOneOutput, err error) {
+					input := &userCreateEvent{}
+
+					err = ctx.WorkflowInput(input)
+
+					if err != nil {
+						return nil, err
+					}
+
+					log.Printf("step-one")
+
+					return &stepOneOutput{
+						Message: "Username is: " + input.Username,
+					}, nil
+				},
+				),
+			},
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error registering workflow: %w", err)
+	}
+
+	var events []client.EventWithMetadata
```

Comment:
@grutt just wanted to get your input on this. I added a new type here to simplify things on the ingest - again I'm a little unsure of the general interface here how this SDK is used.

Reply:
This looks good to me, the existing `Push` method takes args with optional `client.PushOpFunc`s  to build the object internally. Personally,  not a fan of this old pattern and prefer how you implemented.

```
		err := c.Event().Push(
			context.Background(),
			"user:create:simple",
			testEvent,
			client.WithEventMetadata(map[string]string{
				"hello": "world",
			}),
		)
```